src/screens/Classroom/CreateClassroomScreen.js

```diff
@@ -0,0 +1,152 @@
+import 'react-native-gesture-handler';
+import React, { useState, useEffect } from 'react';
+// import { NavigationContainer } from '@react-navigation/native';
+// import { createStackNavigator } from '@react-navigation/stack';
+import {
+  Text, View, StyleSheet, Alert, Button, TextInput,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import Firestore, { firebase } from '@react-native-firebase/firestore';
+// import Post from './Post';
+import { INITIAL_USER_STATE } from '../../components';
+
+const styles = StyleSheet.create({
+  container: {
+    marginTop: 10,
+    marginBottom: 10,
+    marginHorizontal: 10,
+    padding: 10,
+    backgroundColor: '#ffffff',
+    borderRadius: 10,
+  },
+  contentContainer: {
+    paddingTop: 20,
+  },
+  timeText: {
+    fontSize: 11,
+  },
+  topicText: {
+    fontSize: 18,
+    fontWeight: 'bold',
+  },
+  bodyText: {
+    fontSize: 14,
+  },
+});
+
+export default function CreateClassroomScreen({ navigation }) {
+  const [title, setTitle] = useState('');
+  const [initializing, setInitializing] = useState(true);
+  const uid = navigation.getParam('uid', null);
+  const ref = Firestore().collection('users');
+  const [userState, setUserState] = useState(INITIAL_USER_STATE);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  async function getUserData() {
+    try {
+      // If we somehow get to this screen with no uid passed, go back to homescreen
+      if (!uid) navigation.navigate('Home');
+      const doc = await ref.doc(uid).get();
+      const data = doc.data();
+      // Handler for case with nonexistent user entries in Firestore
+      if (!data) {
+        await ref.doc(uid).set({
+          ...userState,
+          createdAt: Firestore.FieldValue.serverTimestamp(),
+          updatedAt: Firestore.FieldValue.serverTimestamp(),
+        });
+      } else {
+        setUserState(data);
+      }
+      if (initializing) setInitializing(false);
+    } catch (e) {
+      setInitializing(false);
+      Alert.alert(
+        e.message,
+      );
+    }
+  }
+
+  useEffect(() => {
+    getUserData();
+  }, [userState]);
+
+  // creates classroomID
+  function makeid(length) {
```

Comment:
While it's not likely that this code will match another randomly generated code, I think we should stick to the unique IDs that Firebase generates for us (because those are guaranteed to be unique within the collection).

Reply:
I had it create a unique code so it would be easy to join - I can definitely rework this however necessary.